components/context_storage/IContextStorage.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
```

Comment:
This "interface" file seems like it shouldn't have to know anything about the various storage options. Can't we change this such that pinecode/weaviate implement an agnostic interface and contain all of the vector store specifics in their respective files?